File: FRHyperLabelDemo/FRHyperLabelDemo/ViewController.m
//
//  ViewController.m
//  FRHyperLabelDemo
//
//  Created by Jinghan Wang on 23/9/15.
//  Copyright © 2015 JW. All rights reserved.
//

#import "ViewController.h"
#import "FRHyperLabel.h"

@interface ViewController ()

@property (nonatomic, weak) IBOutlet FRHyperLabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	
	FRHyperLabel *label = self.label;
	label.numberOfLines = 0;
	
	NSString *string = @"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis blandit eros, sit amet vehicula justo. Nam at urna neque. Maecenas ac sem eu sem porta dictum nec vel tellus.";
	NSDictionary *attributes = @{NSForegroundColorAttributeName: [UIColor blackColor],NSFontAttributeName: [UIFont preferredFontForTextStyle:UIFontTextStyleHeadline]};
	
	NSAttributedString *attributedString = [[NSAttributedString alloc]initWithString:string attributes:attributes];
	
	label.attributedText = attributedString;
	
	NSArray *linkStrings = @[@"Lorem", @"Pellentesque", @"blandit", @"Maecenas"];
	
	for (NSString *linkString in linkStrings) {
		NSRange range = [string rangeOfString:linkString];
		[label setLinkForRange:range withLinkHandler:^{
			NSLog(@"Selected: %@", linkString);
		}];
	}
}

- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

@end

File: FRHyperLabelDemoSwift/FRHyperLabelDemoSwift/ViewController.swift
//
//  ViewController.swift
//  FRHyperLabelDemoSwift
//
//  Created by Jinghan Wang on 3/2/16.
//  Copyright © 2016 JW. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

	@IBOutlet weak var label: FRHyperLabel!
	
	override func viewDidLoad() {
		super.viewDidLoad()
		// Do any additional setup after loading the view, typically from a nib.
		
		label.numberOfLines = 0;
		
		//Step 1: Define a normal attributed string for non-link texts
		let string = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis blandit eros, sit amet vehicula justo. Nam at urna neque. Maecenas ac sem eu sem porta dictum nec vel tellus."
		
		let attributes = [NSForegroundColorAttributeName: UIColor.blackColor(),
			NSFontAttributeName: UIFont.preferredFontForTextStyle(UIFontTextStyleHeadline)]
		
		label.attributedText = NSAttributedString(string: string, attributes: attributes)
		
		//Step 2: Define a selection handler block
		let handler = {
			(hyperLabel: FRHyperLabel!, substring: String!) -> Void in
			let controller = UIAlertController(title: substring, message: nil, preferredStyle: UIAlertControllerStyle.Alert)
			controller.addAction(UIAlertAction(title: "Cancel", style: UIAlertActionStyle.Cancel, handler: nil))
			self.presentViewController(controller, animated: true, completion: nil)
		}
		
		//Step 3: Add link substrings
		label.setLinksForSubstrings(["Lorem", "Pellentesque", "blandit", "Maecenas"], withLinkHandler: handler)
	}

	override func didReceiveMemoryWarning() {
		super.didReceiveMemoryWarning()
		// Dispose of any resources that can be recreated.
	}


}

